flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriterTest.java

```diff
@@ -112,16 +112,20 @@ public void checkLoggedSendTimesAreWithinBounds() throws IOException, Interrupte
         AsyncSinkWriterImpl sink =
                 new AsyncSinkWriterImplBuilder()
                         .context(sinkInitContext)
-                        .maxBatchSize(4)
-                        .delay(100)
+                        .maxBatchSize(2)
+                        .delay(50)
                         .build();
-        for (int i = 0; i < 4; i++) {
-            sink.write(String.valueOf(i));
-        }
+        sink.write(String.valueOf(1));
+        // introduce artificial delay, shouldn't be calculated in send time
+        Thread.sleep(50);
```

Comment:
This doesn't look correct. Can you elaborate why do we need this? Why `sink.flush()` is not waiting for any async things to complete? It looks like actually might be a bug in production code if we have to add `Thread.sleep(x)` to test this.

Reply:
Hi @pnowojski 
`sink.flush()` in fact waits for the async things to complete.
This is in fact why we are adding a sleep. We are testing that SendTime Metric calculation measures time of sending (in this case flush) so we simulate a delay between writing and flushing.
This is an alternative to having an estimate upper bound that is affected by mailbox delays and the reason behind the test flakiness before.

Reply:
Ahh, sorry I completely missed that! LGTM